src/python/library/tritonclient/utils/shared_memory/__init__.py

```diff
@@ -101,27 +58,52 @@ def create_shared_memory_region(triton_shm_name, shm_key, byte_size):
         The unique key of the shared memory object.
     byte_size : int
         The size in bytes of the shared memory region to be created.
+    create_only : bool
+        Whether a shared memory region must be created. If False and
+        a shared memory region of the same name exists, a handle to that
+        shared memory region will be returned and user must be aware that
+        the shared memory size can be different from the size requested.
 
     Returns
     -------
-    shm_handle : c_void_p
+    shm_handle : SharedMemoryRegion
         The handle for the system shared memory region.
 
     Raises
     ------
     SharedMemoryException
         If unable to create the shared memory region.
     """
-
-    shm_handle = c_void_p()
-    _raise_if_error(
-        c_int(
-            _cshm_shared_memory_region_create(
-                triton_shm_name, shm_key, byte_size, byref(shm_handle)
+    shm_handle = SharedMemoryRegion(triton_shm_name, shm_key)
+    # Check whether the region exists before creating it
+    if not create_only:
+        try:
+            shm_handle._mpsm_handle = mpshm.SharedMemory(shm_key)
+            if shm_key not in _key_mapping:
+                _key_mapping[shm_key] = [False, 0]
```

Comment:
```suggestion
                _key_mapping[shm_key] = {"needs_unlink":False, "active_handle_count":0}
```
IIUC this may be more legible if applied throughout the file.